### PR TITLE
Relayer price awareness

### PR DESCRIPTION
**Pull request by ChristianBorst**

### CODE REVIEW

File: solidity/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "test": "npx hardhat test --network localhost",
     "typechain": "npx hardhat typechain",
     "compile-deployer": "npx tsc -p . ; pkg dist/contract-deployer.js --targets node10-linux-x64",
-    "evm_fork": "npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/${ALCHEMY_ID} --fork-block-number 11780000",
+    "evm_fork": "npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/${ALCHEMY_ID} --fork-block-number 12782071",
```
Comment (jkilpatr) [inline solidity/package.json:11]: notice the solidity arbitrary logic test is failing, I suggest we change it's script to use solidity_test_fork and keep evm fork for our new test runner test. That way we can use the pinned block that we updated here in that old test and keep it working.

File: tests/run-all-tests.sh [hunk 2]
```diff
@@ -12,4 +12,6 @@ bash all-up-test.sh VALIDATOR_OUT
 bash all-up-test.sh LONDON
 # An alchemy api key is necessary for this arbitrary logic test, so we are disabling it
 #bash all-up-test.sh ARBITRARY_LOGIC
+# An alchemy api key is necessary for this relay market test, so we are disabling it
+#bash all-up-test.sh RELAY_MARKET
```
Comment (jkilpatr) [inline tests/run-all-tests.sh:16]: this should probably have 'RELAY_MARKET API_KEY' to make it clear how it's intended to be called. Or we could add a bash conditional to only run it when the value is set, probably not needed though.

**Review by jkilpatr — CHANGES_REQUESTED** [review]
Good work! Just needs some changes to correct for clippy, rustfmt and correct the arbitrary logic solidity test.

**Review by jkilpatr — APPROVED** [review]

File: orchestrator/relayer/src/batch_relaying.rs [hunk 3]
```diff
@@ -193,27 +243,47 @@ async fn submit_batches(
...
+                        ethereum_key,
+                    )
+                    .await;
+                    if res.is_err() {
+                        info!("Batch submission failed with {:?}", res);
+                    }
+                } else {
+                    info!(
+                        "Not relaying batch due to it not being profitable: {:?}",
+                        oldest_signed_batch
```
Comment (levicook) [inline orchestrator/relayer/src/batch_relaying.rs:285]: 👋 Might want to include cost and reward in this log statement to improve observability on _why_ it didn't relay.

**Pull request merged**